### PR TITLE
Remove hardcoded functions

### PR DESCRIPTION
**Pull request by solomon-b**
Replaces hard coded functions with custom functions.

### CODE REVIEW

File: src/Kriti/CustomFunctions.hs [hunk 1]
```diff
@@ -105,6 +108,9 @@ toUpperF = parserToFunc $ J.withText "String" $ pure . J.String . T.toUpper
 toTitleF :: KritiFunc
 toTitleF = parserToFunc $ J.withText "String" $ pure . J.String . T.toTitle
 
+escapeUriF :: KritiFunc
+escapeUriF = parserToFunc $ J.withText "String" $ pure . J.String . T.pack . URI.escapeURIString URI.isUnreserved . T.unpack
```
Comment (sordina) [inline src/Kriti/CustomFunctions.hs:112]: 👍

File: src/Kriti/CustomFunctions.hs [hunk 2]
```diff
@@ -144,6 +150,11 @@ concatF = parserToFunc $ J.withArray "Array" \as -> do
       o = J.Object . fold . reverse <$> traverse (J.withObject "Nested Object" pure) l
   a <|> s <|> o
 
+notF :: KritiFunc
+notF = parserToFunc $ J.withBool "Bool" \case
+  False -> pure $ J.Bool True
+  True -> pure $ J.Bool False
```
Comment (sordina) [inline src/Kriti/CustomFunctions.hs:156]: Would truthy operations make sense like `not null == true`?

File: src/Kriti/Eval.hs [hunk 3]
```diff
@@ -118,12 +117,6 @@ evalWith funcMap = \case
       J.Bool True -> eval t1
       J.Bool False -> eval t2
       p' -> throwError $ TypeError src sp $ renderBL $ "'" <> J.encode p' <> "' is not a boolean."
-  Not sp t1 -> do
-    src <- asks fst
-    v1 <- eval t1
-    case v1 of
-      J.Bool p -> pure $ J.Bool $ not p
-      _ -> throwError $ TypeError src sp $ T.pack $ show t1 <> "is not a boolean."
```
Comment (sordina) [inline src/Kriti/Eval.hs:126]: 👍

File: test/data/eval/success/golden/example10.json [hunk 4]
```diff
@@ -1,22 +1,22 @@
 [
     {
-        "name": "Shepard Mccall",
-        "id": 0
+        "id": 0,
+        "name": "Shepard Mccall"
```
Comment (sordina) [inline test/data/eval/success/golden/example10.json:4]: Why does the order change for this?

**Review by sordina — APPROVED** [review]
Looks great @solomon-b  - Really cleans things up.

Only question is why the golden output changed order for some of the fields? Also threw in a question about thruty operations, but that's not a blocker and can stay as is if you prefer.

File: test/data/eval/success/golden/example10.json (hunk 4 above)
Comment (solomon-b) [reply test/data/eval/success/golden/example10.json:4]: I'm not really sure TBH. The order comes from serializing hashmaps as json objects. In one of the later PRs I replace the hashmaps with assoc lists (which should be insert order hash maps once I write a Lift instance for it) so this goes away.

File: src/Kriti/CustomFunctions.hs (hunk 2 above)
Comment (solomon-b) [reply src/Kriti/CustomFunctions.hs:156]: We could do that but it should go in a seperate PR.

**Pull request merged**